### PR TITLE
EVG-7104 check if commit-queue earlier in set-module

### PR DESCRIPTION
**Pull request by ablack12**

### CODE REVIEW

**Review by ybrill — APPROVED** [review]

**Comment by ybrill** [comment]
I still don't understand how the [error in the ticket](https://jira.mongodb.org/browse/EVG-7104?focusedCommentId=3358446&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-3358446) happened, though.

**Comment by ablack12** [comment]
Looked into it: I believe it's because we log the first message at the error level, but then we return the error we got from the legacy client, which is possibly MethodNotAllowed because gimlet can't find results for the route, and so assumes you're not using the correct method, because that route is valid for other methods, but I didn't read the code to know exactly why it went to this

**Comment by ablack12** [comment]
Oh wait I understand what you mean. UpdatePatchModule was _supposed_ to fail, but this error comes from GetPatchModules, which I would expect to work even if the user accidentally used the wrong things. I'm going to look into it a bit more.

**Comment by ablack12** [comment]
Got it! I think we can't assume that patched projects have an identifier, so if they don't gimlet returns MethodNotAllowed because it's not a valid route considering the projectId doesn't exist. I was able to do
`clients/darwin_amd64/evergreen set-module -i 5f650fd49ccd4e13532007d2 -m enterprise` (non-commit-queue patch with the evergreen project) and have it fail before the changes with the MethodNotAllowed, and the after the changes with the expected:
```
module 'enterprise' unknown or not found
known modules includes:
	
could not set specified module: "enterprise"
```

**Comment by ablack12** [comment]
Re-requesting your review since I made a number of changes @ybrill .

**Review by ybrill — APPROVED** [review]

File: operations/patch_module.go [hunk 1]
```diff
@@ -19,12 +18,13 @@ func PatchSetModule() cli.Command {
 		Name:    "patch-set-module",
 		Aliases: []string{"set-module"},
 		Usage:   "update or add module to an existing patch",
-		Flags: mergeFlagSlices(addPatchIDFlag(), addPathFlag(), addModuleFlag(), addYesFlag(), addRefFlag(), addUncommittedChangesFlag(), addPreserveCommitsFlag(
```
[the remark this answers is not in the document]
Comment (ablack12) [reply operations/patch_module.go:22]: good catch!

**Pull request merged**